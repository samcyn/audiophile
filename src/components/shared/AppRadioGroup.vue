<script lang="ts">
// prevent inheritting attributes on root elements
export default {
  inheritAttrs: false,
};
</script>
<script setup lang="ts">
import { useRadioGroup, AppRadioProps } from '../../hooks/useRadio';
import AppRadio from /** // AppRadioGroupProps
 * https://github.com/vuejs/core/issues/4294
 * can't for now import interface in vue3 component
 * it's a bit redundant but hopefully this get resolve
 */ './AppRadio.vue';

type AppRadioGroupProps = {
  label?: string;
  name: string;
  modelValue?: string | number | boolean;
  disabled?: boolean;
  options?: AppRadioProps[];
  defaultValue?: string | number | boolean;
};

const props = withDefaults(defineProps<AppRadioGroupProps>(), {
  label: undefined,
  name: '',
  modelValue: undefined,
  disabled: false,
  options: undefined,
  defaultValue: undefined,
});
const emit = defineEmits(['update:modelValue']);

useRadioGroup(props, emit);
</script>
<template>
  <div role="radiogroup">
    <slot>
      <!-- use options if slot not provided -->
      <AppRadio v-for="option in options" :key="option.label" v-bind="option" />
    </slot>
  </div>
</template>
