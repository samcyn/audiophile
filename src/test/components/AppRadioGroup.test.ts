import { mount } from '@vue/test-utils';
import AppRadioGroup from '../../components/AppRadioGroup.vue';

test('mount AppRadioGroup component', async () => {
  expect(AppRadioGroup).toBeTruthy();

  const wrapper = mount(AppRadioGroup, {
    props: {},
  });

  expect(wrapper.html()).toMatchSnapshot();
});
