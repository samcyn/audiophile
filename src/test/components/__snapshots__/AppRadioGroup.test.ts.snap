// Vitest Snapshot v1

exports[`mount AppRadioGroup component 1`] = `
"<div role=\\"radiogroup\\">
  <!-- use options if slot not provided -->
</div>"
`;
